### PR TITLE
Adding MelonModules (kinda rewriting the compatiblity layer base)

### PR DESCRIPTION
**Pull request by slxdy**

### CODE REVIEW

**Comment by slxdy** [comment]
Just adding some extra stuff I need before actually writing the Modules base

**Comment by slxdy** [comment]
Now im gonna turn the start screen into a MelonModule : )

File: Dependencies/Il2CppAssemblyGenerator/Core.cs [hunk 1]
```diff
@@ -2,11 +2,11 @@
 using System.IO;
 using System.Net;
 using MelonLoader.Il2CppAssemblyGenerator.Packages;
+using MelonLoader.Modules;
 
 namespace MelonLoader.Il2CppAssemblyGenerator
 {
-
-	internal static class Core
+	internal class Core : MelonModule
```
Comment (SamboyCoding) [inline Dependencies/Il2CppAssemblyGenerator/Core.cs]: This line starts with a tab while the rest start with spaces

File: Dependencies/MelonStartScreen/UnityEngine/CoreModule/ImageConversion.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 using System;
 using System.Runtime.InteropServices;
 using UnhollowerMini;
+using static MelonLoader.MelonStartScreen.Core;
```
Comment (SamboyCoding) [inline Dependencies/MelonStartScreen/UnityEngine/CoreModule/ImageConversion.cs]: Can we not do this, just use Core.Logger further down

**Review by SamboyCoding — CHANGES_REQUESTED** [review]
Looks good aside from two very minor details

File: Dependencies/Il2CppAssemblyGenerator/Core.cs (hunk 1 above)
Comment (slxdy) [reply Dependencies/Il2CppAssemblyGenerator/Core.cs]: ... you fr

File: Dependencies/MelonStartScreen/UnityEngine/CoreModule/ImageConversion.cs (hunk 2 above)
Comment (slxdy) [reply Dependencies/MelonStartScreen/UnityEngine/CoreModule/ImageConversion.cs]: Fine...

File: Dependencies/Il2CppAssemblyGenerator/Core.cs (hunk 1 above)
Comment (slxdy) [reply Dependencies/Il2CppAssemblyGenerator/Core.cs]: https://github.com/LavaGang/MelonLoader/pull/197/commits/9aece8a968fd9dd195cde12bf732747f3d65204d

File: Dependencies/MelonStartScreen/UnityEngine/CoreModule/ImageConversion.cs (hunk 2 above)
Comment (slxdy) [reply Dependencies/MelonStartScreen/UnityEngine/CoreModule/ImageConversion.cs]: https://github.com/LavaGang/MelonLoader/pull/197/commits/cb08a9238cbf255958dfb97e9317170df708569b

**Review by SamboyCoding — APPROVED** [review]

**Pull request merged**